plugins/inputs/win_wmi/win_wmi.go

```diff
@@ -0,0 +1,249 @@
+//go:build windows
+// +build windows
+
+package win_wmi
+
+import (
+	_ "embed"
+	"fmt"
+	"runtime"
+	"strings"
+	"sync"
+
+	ole "github.com/go-ole/go-ole"
+	"github.com/go-ole/go-ole/oleutil"
+	"github.com/influxdata/telegraf"
+	"github.com/influxdata/telegraf/filter"
+	"github.com/influxdata/telegraf/internal"
+	"github.com/influxdata/telegraf/plugins/inputs"
+)
+
+//go:embed sample.conf
+var sampleConfig string
+
+// Query struct
+type Query struct {
+	query                string
+	Namespace            string   `toml:"namespace"`
+	ClassName            string   `toml:"class_name"`
+	Properties           []string `toml:"properties"`
+	Filter               string   `toml:"filter"`
+	TagPropertiesInclude []string `toml:"tag_properties"`
+	tagFilter            filter.Filter
+}
+
+// Wmi struct
+type Wmi struct {
+	Queries []Query `toml:"query"`
+	Log     telegraf.Logger
+}
+
+// S_FALSE is returned by CoInitializeEx if it was already called on this thread.
+const sFalse = 0x00000001
+
+func oleInt64(item *ole.IDispatch, prop string) (int64, error) {
+	v, err := oleutil.GetProperty(item, prop)
+	if err != nil {
+		return 0, err
+	}
+	defer func() { _ = v.Clear() }()
+
+	i := v.Val
+	return i, nil
+}
+
+// Init function
+func (s *Wmi) Init() error {
+	if err := compileInputs(s); err != nil {
+		return err
+	}
+	return nil
+}
+
+// SampleConfig function
+func (s *Wmi) SampleConfig() string {
+	return sampleConfig
+}
+
+func compileInputs(s *Wmi) error {
+	buildWqlStatements(s)
+	if err := compileTagFilters(s); err != nil {
+		return err
+	}
+	return nil
+}
+
+func compileTagFilters(s *Wmi) error {
+	var err error
+	for i, q := range s.Queries {
```

Comment:
Please move the declaration as-close-as-possible to the use to minimize the scope of the declaration and to avoid potential side-effects
```suggestion
	for i, q := range s.Queries {
		var err error
```